### PR TITLE
[alpine] Upgrade 3.11.6 -> 3.12.1, [docs] Customization

### PR DESCRIPTION
**Pull request by Nuru**
## what
- Update Alpine-based Geodesic from 3.11.6 to 3.12.1, closes #592 
- Add documentation on customization options
- Improve man page generation

## why
- Stay up to date with Alpine. In particular, Alpine does not maintain a lot support for old versions
- Geodesic has powerful customization options, but they are in no way intuitive or obvious. People need documentation to know about them and use them.
- The use of dollar signs (e.g. for shell variables) was confusing the formatter because it sometimes created constructs that looked like LaTex macros, causing mangled formatting on the output.

### CODE REVIEW

File: docs/customization.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+title: Customization(5) | Geodesic
+author:
+- Nuru
+date: March 2019
```
Comment (aknysh) [inline docs/customization.md:5]: correct date?

File: os/debian/Dockerfile.debian [hunk 2]
```diff
@@ -85,7 +93,7 @@ RUN curl -1sLf 'https://dl.cloudsmith.io/public/cloudposse/packages/cfg/setup/ba
 
 # Insall Google package repo
```
Comment (aknysh) [inline os/debian/Dockerfile.debian:94]:
```suggestion
# Install Google package repo
```

File: os/debian/Dockerfile.debian [hunk 3]
```diff
@@ -85,7 +93,7 @@ RUN curl -1sLf 'https://dl.cloudsmith.io/public/cloudposse/packages/cfg/setup/ba
...
 RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
-RUN curl -1sLf https://packages.cloud.google.com/apt/doc/apt-key.gpg | APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=true apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+RUN curl -1sLf 'https://packages.cloud.google.com/apt/doc/apt-key.gp'g | APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=true apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
```
Comment (aknysh) [inline os/debian/Dockerfile.debian:96]: gp'g - is this correct?

File: rootfs/usr/local/bin/docs [hunk 4]
```diff
@@ -29,9 +29,12 @@ tasks:
       mkdir -p ${man_pages}
       for md in $(find /usr/share/docs -type f -name '*.md'); do
         man=${man_pages}/$(basename $md .md).$section
-        # Verbatim text (i.e. between backticks, `…`) doesn’t show in any particular way
-        sed 's/`/**/g' < $md | sed -E 's/^#(#+ )/\1/g' | \
-          pandoc -s -f markdown -t man | sed 's/\\\[en\]/-/' > $man
+        # Verbatim text (i.e. between backticks, `…`) doesn’t show in any particular way, so we bold it
+        # by using "**", but we need to preserve code fence triple backticks "```".
```
Comment (aknysh) [inline rootfs/usr/local/bin/docs:33]:
```suggestion
        # by using "**", but we need to preserve code hence triple backticks "```".
```

**Review by aknysh — COMMENTED** [review]
please see comments

File: docs/customization.md (hunk 1 above)
Comment (Nuru) [reply docs/customization.md:5]: Yes, this is copied from the PR.

File: os/debian/Dockerfile.debian (hunk 2 above)
Comment (Nuru) [reply os/debian/Dockerfile.debian:94]: Will fix, thanks

File: os/debian/Dockerfile.debian (hunk 3 above)
Comment (Nuru) [reply os/debian/Dockerfile.debian:96]: Will fix, thanks

File: rootfs/usr/local/bin/docs (hunk 4 above)
Comment (Nuru) [reply rootfs/usr/local/bin/docs:33]: No, the triple backticks is called a "code fence".

**Review by aknysh — APPROVED** [review]

**Pull request merged**